cmd/portal/analytic_report.go

```diff
@@ -146,14 +124,24 @@ var cmdAnalyticReport = &cobra.Command{
 			}, nil
 		}
 
+		period, err := binder.GetRequiredString(cmd, ArgAnalyticPeriod)
+		if err != nil {
+			return err
+		}
+		parser := analytic.NewPeriodicalArgumentParser()
+		periodicalType, date, err := parser.Parse(period)
+		if err != nil {
+			return err
+		}
+
 		var data *analyticlib.ReportData
 		dbPool := db.NewPool()
 		switch reportType {
 		case analytic.ReportTypeUserWeeklyReport:
-			year, week, err := getISOWeek()
-			if err != nil {
-				return err
+			if periodicalType != periodical.Weekly {
```

Comment:
I have another idea of  changing the CLI interface.

```sh
go run ./cmd/portal analytic report user --period=last-week
go run ./cmd/portal analytic report project --period=last-month
```

Then we eliminate the problem of duplicating the period in the report type argument.

Reply:
Updated!